### ISSUE TITLE
Stopping emd during sync leads to data corruption

### ISSUE THREAD

**Issue by mdyring** [e-money/em-ledger#4]
Stopping emd during sync leads to data corruption. So far only reproducible on Linux, not MacOS.

Syncing:
```
...
I[2019-12-28|12:41:25.327] Committed state                              module=state height=586 txs=0 appHash=39B90E4E3358A5639BC84FE39BCC2EC280EABC191EFDD0AD81EAC707A460690A
I[2019-12-28|12:41:25.339] Endblock: Block 587 was proposed by D48BB5DA468ED61E27E52414CD11BE67983D128A module=emz 
I[2019-12-28|12:41:25.340] Executed block                               module=state height=587 validTxs=0 invalidTxs=0
E[2019-12-28|12:41:25.342] Connection failed @ recvRoutine (reading byte) module=p2p peer=277ead1312fc456d98639b3e4fc8c37dc4df27c8@167.86.70.194:26656 conn=MConn{167.86.70.194:26656} err="read tcp 116.202.108.95:39516->167.86.70.194:26656: use of closed network connection"
E[2019-12-28|12:41:25.342] Stopping peer for error                      module=p2p peer="Peer{MConn{167.86.70.194:26656} 277ead1312fc456d98639b3e4fc8c37dc4df27c8 out}" err="read tcp 116.202.108.95:39516->167.86.70.194:26656: use of closed network connection"
E[2019-12-28|12:41:25.343] Not stopping ConsensusState -- have not been started yet module=consensus impl=ConsensusState
I[2019-12-28|12:41:25.343] exiting...                                   module=main
```

Crash:
```
emd start
I[2019-12-28|12:37:02.584] starting ABCI with Tendermint                module=main 
panic: failed to load Store: wanted to load target 525 but only found up to 0

goroutine 1 [running]:
emoney.NewApp(0x1555fe0, 0xc000a0caa0, 0x1567e20, 0xc000010230, 0xc0000aff00, 0x4)
        /home/emoney/em-ledger/app.go:192 +0x438b
main.newAppCreator.func1(0x1555fe0, 0xc000a0caa0, 0x1567e20, 0xc000010230, 0x0, 0x0, 0xc0009f5080, 0x9c)
        /home/emoney/em-ledger/cmd/emd/main.go:63 +0x52
github.com/cosmos/cosmos-sdk/server.startInProcess(0xc0000aff00, 0xc0003591e0, 0x1d, 0x0, 0x0)
        /home/emoney/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.37.3/server/start.go:142 +0x134
github.com/cosmos/cosmos-sdk/server.StartCmd.func1(0xc00046e280, 0x1fac4d0, 0x0, 0x0, 0x0, 0x0)
        /home/emoney/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.37.3/server/start.go:65 +0xb4
github.com/spf13/cobra.(*Command).execute(0xc00046e280, 0x1fac4d0, 0x0, 0x0, 0xc00046e280, 0x1fac4d0)
        /home/emoney/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc000132000, 0x2, 0xc0008661a0, 0x11ba06d)
        /home/emoney/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /home/emoney/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/tendermint/tendermint/libs/cli.Executor.Execute(0xc000132000, 0x1390330, 0x3, 0xc0008d7a80)
        /home/emoney/go/pkg/mod/github.com/tendermint/tendermint@v0.32.6/libs/cli/setup.go:89 +0x3c
main.main()
        /home/emoney/em-ledger/cmd/emd/main.go:55 +0x398
```

**Comment by mdyring** [comment]
Looks related to https://github.com/tendermint/tendermint/issues/4069 which is fixed in Tendermint v0.32.9:

https://github.com/tendermint/tendermint/blob/v0.32/CHANGELOG.md#v0329

**Comment by mdyring** [comment]
Fixed in 41c32b9bcaec81492e0601474b7620d178259b2d.

Was due to pruning options not being honored, is "syncable" now.
